### PR TITLE
Add dummy helm chart for installing Application

### PR DESCRIPTION
**Pull request by martinhaus**
Adds an internal helm chart that bundles a single object of ArgoCD Application. This approach should avoid problems with using Kubernetes Terraform provider, specifically issues with running terraform plan without the cluster already spun up, or without installing the ArgoCD CRDs in a prior terraform run.

Bundling the Kubernetes object avoids comparing the state of the Kubernetes resource with live infrastructure, so no errors should be produced during plan time.

### CODE REVIEW

File: application.tf [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
-      }
-      "syncPolicy" = var.argo_sync_policy
-      "info"       = var.argo_info
     }
-  }
+    "destination" : {
+      "server" : var.argo_destionation_server
+      "namespace" : var.k8s_namespace
+    }
+    "syncPolicy" : var.argo_sync_policy
```
Comment (Thumbiceq) [inline application.tf:18]: heads up:
var.argo_sync_policy should have new format I believe. 

Right now: { "syncOptions" = ["CreateNamespace=true"] }
After this change (probably smth like this): { "syncOptions" : ["CreateNamespace=true"] }

Comment (martinhaus) [reply application.tf:18]: Both options produce the same result

File: helm/argocd-application/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+appVersion: 1.16.0
```
Comment (dojci) [inline helm/argocd-application/Chart.yaml:23]: Is this variable necessary ?

File: helm/argocd-application/templates/_helpers.tpl [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "argocd_application.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "argocd_application.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
```
Comment (dojci) [inline helm/argocd-application/templates/_helpers.tpl:62]: Please remove unused code

File: helm/argocd-application/templates/application.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{ with .Values.syncPolicy }}
+  syncPolicy:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{ with .Values.info }}
+  info:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
```
Comment (dojci) [inline helm/argocd-application/templates/application.yaml:24]: Why don't we use whole blob for configuration template ?

**Review by dojci — CHANGES_REQUESTED** [review]

File: helm/argocd-application/templates/application.yaml (hunk 4 above)
Comment (martinhaus) [reply helm/argocd-application/templates/application.yaml:24]: It's easier to have it in this format when we want to reuse the same blob for Kubernetes manifest object. It also handles all of the possible cases for this object so unless the object definition changes, it should have the same result.

**Review by dojci — APPROVED** [review]

**Pull request merged**